utils/ort/src/main/kotlin/storage/S3FileStorage.kt

```diff
@@ -62,29 +55,34 @@ class S3FileStorage(
     /** Custom endpoint to perform AWS API Requests */
     private val customEndpoint: String? = null,
 
+    /** Whether to enable path-style access or not. Defaults to false. */
+    private val pathStyleAccess: Boolean = false,
+
     /** The AWS secret for the access key. */
-    private val secretAccessKey: String? = null
+    private val secretAccessKey: String? = null,
 ) : FileStorage {
     private val s3Client: S3Client by lazy {
-        val provider = if (accessKeyId != null && secretAccessKey != null) {
-            StaticCredentialsProvider.create(AwsBasicCredentials.create(accessKeyId, secretAccessKey))
-        } else {
-            null
-        }
-
-        if (awsRegion != null && provider != null) {
-            S3Client.builder()
-                .region(Region.of(awsRegion))
-                .credentialsProvider(provider)
-                .endpointOverride(if (customEndpoint != null) URI(customEndpoint) else null)
-                .build()
-        } else {
+        S3Client.builder().apply {
             if (awsRegion != null) {
-                S3Client.builder().region(Region.of(awsRegion)).build()
-            } else {
-                S3Client.create()
+                region(Region.of(awsRegion))
```

Comment:
I like the `apply`-style refactoring, but I believe it deserves to be done in a preceding "non-functional refactoring" commit, to better highlight the actual feature-change you're doing later.

Also the separate `if`-blocks should be separated by empty lines for clarity.